packages/studio-be/src/studio/cloud/oauth-client.ts

```diff
@@ -0,0 +1,41 @@
+import axios, { AxiosInstance } from 'axios'
+import qs from 'qs'
+import VError from 'verror'
+
+export class OAuthClient {
+  private axios: AxiosInstance
+  constructor(props: { oauthEndpoint: string }) {
+    this.axios = axios.create({ baseURL: props.oauthEndpoint })
+  }
+
+  public async getAccessToken(props: { clientId: string; clientSecret: string }): Promise<string | undefined> {
+    const { clientId, clientSecret } = props
+
+    try {
+      const { data } = await this.axios.post(
+        '/',
+        qs.stringify({
+          client_id: clientId,
+          client_secret: clientSecret,
+          grant_type: 'client_credentials',
+          scope: 'nlu'
```

Comment:
only nlu ?

Reply:
Yeah it's been working fine so far with that, we can parametrize it in the future. I'd rather keep it like this for now.